hazelcast/src/main/java/com/hazelcast/internal/dynamicconfig/ClusterWideConfigurationService.java

```diff
@@ -272,81 +252,43 @@ private IdentifiedDataSerializable cloneConfig(IdentifiedDataSerializable config
-        } else if (newConfig instanceof QueueConfig) {
-            QueueConfig queueConfig = (QueueConfig) newConfig;
-            currentConfig = queueConfigs.putIfAbsent(queueConfig.getName(), queueConfig);
-        } else if (newConfig instanceof ReliableTopicConfig) {
-            ReliableTopicConfig reliableTopicConfig = (ReliableTopicConfig) newConfig;
-            currentConfig = reliableTopicConfigs.putIfAbsent(reliableTopicConfig.getName(), reliableTopicConfig);
-        } else if (newConfig instanceof CacheSimpleConfig) {
-            CacheSimpleConfig cacheSimpleConfig = (CacheSimpleConfig) newConfig;
-            currentConfig = cacheSimpleConfigs.putIfAbsent(cacheSimpleConfig.getName(), cacheSimpleConfig);
-            if (currentConfig == null) {
-                listener.onConfigRegistered(cacheSimpleConfig);
-            }
-        } else if (newConfig instanceof FlakeIdGeneratorConfig) {
-            FlakeIdGeneratorConfig config = (FlakeIdGeneratorConfig) newConfig;
-            currentConfig = flakeIdGeneratorConfigs.putIfAbsent(config.getName(), config);
-        } else if (newConfig instanceof PNCounterConfig) {
-            PNCounterConfig config = (PNCounterConfig) newConfig;
-            currentConfig = pnCounterConfigs.putIfAbsent(config.getName(), config);
-        } else if (newConfig instanceof DataConnectionConfig) {
-            DataConnectionConfig config = (DataConnectionConfig) newConfig;
-            currentConfig = dataConnectionConfigs.putIfAbsent(config.getName(), config);
-            if (currentConfig == null) {
-                nodeEngine.getDataConnectionService().createConfigDataConnection(config);
-            }
-        } else if (newConfig instanceof WanReplicationConfig) {
-            WanReplicationConfig config = (WanReplicationConfig) newConfig;
-            currentConfig = wanReplicationConfigs.putIfAbsent(config.getName(), config);
-            if (currentConfig == null) {
-                nodeEngine.getWanReplicationService().addWanReplicationConfig(config);
+        if ((newConfig instanceof MultiMapConfig) || (newConfig instanceof MapConfig)
+                || (newConfig instanceof CardinalityEstimatorConfig) || (newConfig instanceof RingbufferConfig)
+                || (newConfig instanceof ListConfig) || (newConfig instanceof SetConfig)
+                || (newConfig instanceof ReplicatedMapConfig) || (newConfig instanceof TopicConfig)
+                || (newConfig instanceof ExecutorConfig) || (newConfig instanceof DurableExecutorConfig)
+                || (newConfig instanceof ScheduledExecutorConfig) || (newConfig instanceof QueueConfig)
+                || (newConfig instanceof ReliableTopicConfig) || (newConfig instanceof CacheSimpleConfig)
+                || (newConfig instanceof FlakeIdGeneratorConfig) || (newConfig instanceof PNCounterConfig)
+                || (newConfig instanceof DataConnectionConfig) || (newConfig instanceof WanReplicationConfig)) {
```

Comment:
I'm not a fan of this huge conditional statement - I feel that the original layout was more understandable (albeit more LOC, but that's not really important over code readability). If we do keep this approach, I think it makes more sense to have a collection of supported configs (use version map?) that we use for checking instead.